clusterloader2/pkg/measurement/common/metrics_server_measurement.go

```diff
@@ -0,0 +1,345 @@
+			}
+
+		}
+	}
+}
+
+func (m *metricsServerLatencyMeasurement) retrieveMetricsServerLatencyMetrics(kubeConf *rest.Config, metricsServerPodName string) (*measurementutil.Histogram, error) {
+	metricsServerHist := measurementutil.NewHistogram(nil)
+	output, err := m.getMetricsFromMS(kubeConf, metricsServerPodName)
+	if err != nil {
+		return metricsServerHist, err
+	}
+	samples, err := measurementutil.ExtractMetricSamples(output)
+	if err != nil {
+		return metricsServerHist, err
+	}
+	for _, sample := range samples {
+		if isMetricsServerLatencySample(sample) {
+			measurementutil.ConvertSampleToHistogram(sample, metricsServerHist)
+		}
+	}
+	return metricsServerHist, nil
+}
+
+func isMetricsServerLatencySample(sample *model.Sample) bool {
+	name, nameOk := sample.Metric[model.MetricNameLabel]
+	group, groupOK := sample.Metric["group"]
+	resource, resourceOk := sample.Metric["resource"]
+	scope, scopeOk := sample.Metric["scope"]
+
+	if nameOk &&
+		groupOK &&
+		resourceOk &&
+		scopeOk &&
+		name == metricsServerRequestDurationMetricName &&
+		group == metricsServerRequestDurationGroup &&
+		resource == metricsServerRequestDurationResource &&
+		scope == metricsServerScope {
+		return true
+	}
+
+	return false
+}
+
+func (m *metricsServerLatencyMeasurement) getMetricsFromMS(kubeConf *rest.Config, metricsServerPodName string) (string, error) {
+	out, err := proxyRequestToPod(kubeConf, m.namespace, metricsServerPodName, "https", 4443, "/metrics")
+	if err != nil {
+		return "", err
+	}
+
+	return string(out), nil
+}
+
+func proxyRequestToPod(config *rest.Config, namespace, podName, scheme string, port int, path string) ([]byte, error) {
+	cancel, err := setupForwarding(config, namespace, podName)
+	defer cancel()
+	if err != nil {
+		return nil, err
+	}
+	var query string
+	if strings.Contains(path, "?") {
+		elm := strings.SplitN(path, "?", 2)
+		path = elm[0]
+		query = elm[1]
+	}
+	reqURL := url.URL{Scheme: scheme, Path: path, RawQuery: query, Host: fmt.Sprintf("127.0.0.1:%d", port)}
+	resp, err := sendRequest(config, reqURL.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+func setupForwarding(config *rest.Config, namespace, podName string) (cancel func(), err error) {
+	hostIP := strings.TrimLeft(config.Host, "https://")
+
+	trans, upgrader, err := spdy.RoundTripperFor(config)
+	if err != nil {
+		return noop, err
+	}
+
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: trans}, http.MethodPost, &url.URL{Scheme: "https", Path: fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, podName), Host: hostIP})
+
+	var berr, bout bytes.Buffer
+	buffErr := bufio.NewWriter(&berr)
+	buffOut := bufio.NewWriter(&bout)
+
+	stopCh := make(chan struct{})
+	readyCh := make(chan struct{})
+
+	fw, err := portforward.New(dialer, []string{"4443:4443"}, stopCh, readyCh, buffOut, buffErr)
```

Comment:
This port also needs to be updated. Check this code https://github.com/kubernetes-sigs/metrics-server/pull/730/files#diff-9e762d0df1502ee31a1de1a3bc67be6c2707ec3e6117be3147907a60ef5fb854R300